src/IotCentral/IotCentral/IotCentral/NewAzureRmIotCentralApp.cs

```diff
@@ -112,7 +113,22 @@ public override void ExecuteCmdlet()
 
         private string GetAppSkuName()
         {
-            return this.Sku ?? PSIotCentralAppSku.S1.ToString();
+            if (string.IsNullOrEmpty(this.Sku))
```

Comment:
````
        [Parameter(
            Mandatory = false,
            ValueFromPipelineByPropertyName = true,
            HelpMessage = "Pricing tier for IoT Central applications. Default value is S1.")]
        [PSArgumentCompleter("S1")]
        [ValidateNotNullOrEmpty]
        public string Sku { get; set; }
````
It is a little strange that a optional parameter is to be validated not null or empty.
And it is impossible to reach the branch of throwing PSArgumentException.
Is it added for legacy codes?

Reply:
I'v tried, with out this validation, it will throw null reference exception when Sku is not passed in. previous code also have this "this.Sku ??". Property definition can not prevent null value pass in.